iree/compiler/Dialect/Shape/Plugins/XLA/XlaHloShapeBuilder.cpp

```diff
@@ -308,6 +308,128 @@ Value rewriteSelectOp(RankedShapeType resultShape, SelectOp selectOp,
   return builder.create<MakeRankedShapeOp>(loc, resultShape, dynamicDims);
 }
 
+Value rewriteTorchIndexSelect(RankedShapeType resultShape,
+                              TorchIndexSelectOp torchIndexSelectOp,
+                              OpBuilder &builder) {
+  if (!torchIndexSelectOp) return nullptr;
+  auto loc = torchIndexSelectOp.getLoc();
+
+  int64_t resultShapeRank = resultShape.getRank();
+  auto paramsType =
+      torchIndexSelectOp.input().getType().cast<RankedTensorType>();
+  auto indicesType =
+      torchIndexSelectOp.index().getType().cast<RankedTensorType>();
+  if (!paramsType || !indicesType) {
```

Comment:
Since you are checking for nullptr, what you want above is dyncast<>, not cast<>. cast will never return nullptr and will assert fail on a mismatched type.

Reply:
done